src/main.cpp

```diff
@@ -97,7 +97,10 @@ int main(int argc, char** argv) {
     try {
       if (!limit_arg.empty()) {
         // Internally timeout is in milliseconds.
-        cl_args.timeout = 1000 * std::stof(limit_arg);
+        const auto timeout = std::chrono::duration<double, std::milli>(
+          1000 * std::stof(limit_arg));
+        cl_args.timeout =
+          std::chrono::duration_cast<std::chrono::milliseconds>(timeout);
```

Comment:
I'm not sure why an intermediate variable is necessary here. Wouldn't  `cl_args.timeout = std::chrono::milliseconds(1000 * std::stof(limit_arg))` do the trick in the same way?

Reply:
`std::chrono` library is very cautious not to lose precision during conversions (at least unless explicitly asked to), so `std::chrono::milliseconds` cannot be constructed from a floating point value. This could be dealt with using a cast
`cl_args.timeout = std::chrono::milliseconds(static_cast<int>(1000 * std::stof(limit_arg)))` and I can change the PR if you prefer this version.

Reply:
Done.